pkgs/tools/misc/libbitcoin/libbitcoin-protocol.nix

```diff
@@ -0,0 +1,38 @@
+{ stdenv, lib, fetchFromGitHub, pkgconfig, autoreconfHook
+, boost, libbitcoin, secp256k1, zeromq }:
+
+let
+  pname = "libbitcoin-protocol";
+  version = "3.4.0";
+
+in stdenv.mkDerivation {
+  name = "${pname}-${version}";
+
+  src = fetchFromGitHub {
+    owner = "libbitcoin";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "1wrlzani3wdjkmxqwjh30i8lg3clrzwcx2di7c9sdpnsbda985gb";
+  };
+
+  nativeBuildInputs = [ autoreconfHook pkgconfig ];
+  buildInputs = [ libbitcoin secp256k1 ];
+  propagatedBuildInputs = [ zeromq ];
```

Comment:
Why is `zeromq` propagated?

Reply:
Because otherwise when compiling `libbitcoin-client` I get:

```
checking for libbitcoin-protocol >= 3.4.0... no
configure: error: Package requirements (libbitcoin-protocol >= 3.4.0) were not met:
Package 'libzmq', required by 'libbitcoin-protocol', not found
Consider adjusting the PKG_CONFIG_PATH environment variable if you
installed software in a non-standard prefix.
Alternatively, you may set the environment variables bitcoin_protocol_CFLAGS
and bitcoin_protocol_LIBS to avoid the need to call pkg-config.
See the pkg-config man page for more details.
```

Reply:
I could add a `zeromq` parameter at the beginning of the `libbitcoin-client` derivation, but it seems like `zeromq` is a direct dependency of `libbitcoin-protocol` instead, so I thought this makes more sense.

Reply:
Please keep it as is.